### DISCUSSION TITLE
(Amazon Lightsail) Server crashes when uploading files

### DISCUSSION THREAD

**Discussion by AsenCME** [pocketbase/pocketbase#4051] [Q&A]
Hi!

I'm using Pocketbase as a systemd service (as per the instructions in step 4 "Going to production"). It starts up and works well while browsing. One of my collections has an "images" attribute which also creates thumbs of size 500x500. 

<img width="644" alt="Screenshot 2023-12-30 at 14 03 58" src="https://github.com/pocketbase/pocketbase/assets/18069689/e6804fd1-9dd5-4918-bbe9-a58678d9b5db">

Whenever I add to this collection 3 images of approximately 3.3MB, my server crashes with no logs and I have to restart it. Later, when I log back in, all the images are there and the thumbs are created but as soon as I add a record, the server stops working and I can't even ssh into it. 

Pocketbase is running on Amazon Linux 2023 in Lightsail with 1GB RAM, 2vCPUs and 40GB of storage.

**Comment by ganigeorgiev** [comment]
Unfortunately the description of the issue is very vague and I'm not sure that I'll be able to help you debug it. I'm currently testing locally with uploading 20x~3-5mb images in a single request and it works fine for me.

> my server crashes with no logs and I have to restart it

What do you mean by "crashes"?
If the process is killed then the systemd should restart it automatically.

For logs check the configured stderr output file (if you use the same systemd config from the docs, that should be `/root/pb/errors.log`).

Alternatively you can also try checking the status of the service with `systemctl status pocketbase` and/or explore the journal logs with something like `journalctl -u pocketbase`.

Additionally:
- Is PocketBase the only running application on the server?
- Do you have custom hooks code?
- What PocketBase version do you use? Note that with v0.20.0 we added a "rate limiter" for the thumbs generation (see https://github.com/pocketbase/pocketbase/issues/3759).

If the cause of the "crash" is OOM, you can try to set `GOMEMLIMIT=500MiB` env variable before starting the executable.

**Reply by ganigeorgiev** [reply]
To assist with the debugging, you can also enable the `--dev` flag when starting the executable and/or change the "Min log level" from the _Admin UI > Settings > Logs > Logs settings_ popup to `-4` and check in the Admin UI listing if there is any debug log related to your "crash".

**Reply by emrahamet** [reply]
@ganigeorgiev 
Have you ever tested what the minimum GOMEMLIMIT should be?

**Reply by ganigeorgiev** [reply]
@emrahamet It is difficult to say and it will depend on your use case, the number of expected concurrent requests, etc. You don't have to set the env variable if you are not experiencing OOM errors.

1GB ram usually should be OK for small projects as long as you don't have to generate thumbs for very large images.
OR alternatively, you can also enable disk swap to compensate if you can't upgrade your RAM.

For benchmarks you can explore https://github.com/pocketbase/benchmarks (_there are also tests for the Fly VM with 256MB_).
